### ISSUE TITLE
float(mpz(int)) != float(int)

### ISSUE THREAD

**Issue by skirpichev** [aleaxit/gmpy#533]
An example (assuming CPython float's are IEEE 754 doubles):
```pycon
>>> x=225_188_150_488_381_457
>>> float(x)
2.2518815048838147e+17
>>> float(mpz(x))
2.2518815048838144e+17
```
I didn't test too much edge cases, but I would expect that overflow conditions also could be slightly different.

On another hand, conversion to mpfr (in appropriate context) and then to a float - works as expected:
```pycon
>>> set_context(ieee(64))
>>> float(mpfr(x))
2.2518815048838147e+17
>>> float.fromhex(f"{mpfr(x):a}")
2.2518815048838147e+17
```

**Comment by casevh** [comment]
I have the fix for float(mpz(x)). I also checked float(mpq(x,1)). It returns the correct value. But it does not raise an OverflowError if the mpq is too large. Should I change float(mpq(large value)) to raise OverflowError instead of returning inf?

**Comment by skirpichev** [comment]
> I have the fix for float(mpz(x)).

I would guess, you are thinking about using mpfr_set_q(), just as in the GMPy_MPQ_Float_Slot() ;)  Seems as a workaround for the gmpy2.  But I think bug is in mpz_get_d().

> But it does not raise an OverflowError if the mpq is too large.

I think we should be consistent with the CPython, where OverflowError is raised.

**Comment by casevh** [comment]
mpz_get_d doesn't support round to nearest. It only supports rounding towards 0.

I'm using mpff_set_z to fix float(mpz)

**Comment by skirpichev** [comment]
> mpz_get_d doesn't support round to nearest.

Indeed.  My bad, I wasn't checked docs, assuming the default rounding mode is same as in C standard.

Then this does make sense.

**Comment by skirpichev** [comment]
It seems fixed by 436d695d, though I'm not sure it's a correct fix.

I think we should make computation in ieee(64) context, rather than in the current.  I'll work on this.  I will also add bulk tests for ``float(mpz(int)) == float(int)``.
